pkgs/by-name/cr/cryptpad/package.nix

```diff
@@ -18,18 +18,55 @@ let
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
-  onlyoffice_build =
+  onlyoffice_oldfetch =
     rev: hash:
     fetchFromGitHub {
       inherit rev hash;
       owner = "cryptpad";
       repo = "onlyoffice-builds";
     };
+
+  # New method for v7+ versions that use ZIP releases
+  onlyoffice_newfetch =
+    version: hash:
+    fetchurl {
+      url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
+      inherit hash;
+    };
+
   onlyoffice_install = oo: ''
     oo_dir="$out_cryptpad/www/common/onlyoffice/dist/${oo.subdir}"
-    cp -a "${onlyoffice_build oo.rev oo.hash}/." "$oo_dir"
-    chmod -R +w "$oo_dir"
-    echo "${oo.rev}" > "$oo_dir/.commit"
+    ${
+      if oo ? "version" then
+        ''
+          # New method for v7+ versions
+          mkdir -p "$oo_dir"
+          cd "$oo_dir"
+          unzip ${onlyoffice_newfetch oo.version oo.hash}
```

Comment:
So I don't see that in the cryptpad install script, but with this we end up with this path:
```
./www/common/onlyoffice/dist/v8/sdkjs/common/serviceworker/document_editor_service_worker.js
```
and after fixing the tests I see a 404 error for this path:
```
vm-test-run-cryptpad> machine # [   17.796145] cryptpad[653]: ["INFO","2025-08-24T02:38:13.244Z","HTTP_404","/common/onlyoffice/dist/v8/document_editor_service_worker.js"]
```
(so same path without `sdkjs/common/serviceworker`)

I think this will need manual testing (or perhaps adding a case to selenium?) and fixing that path too :/